### DISCUSSION TITLE
How do you pull in Front Matter into Statblocks

### DISCUSSION THREAD

**Discussion by BobHood** [Obsidian-TTRPG-Community/fantasy-statblocks#121] [Q&A]
I've got all the monsters for the system I'm running in a .csv and I'm importing them using a Handlebar Template into their own notes (by the monster). All the data from the .CSV is being pulled into the Frontmatter with variables. I've created a custom Statblock layout and set it as default in the plugin. 
Now I'm having a problem pulling the data from the Frontmatter into the statblock. I'm posting the entire note example below, and the json for the custom layout for "Hyperborea 3E Monster" below that...

Any Help would be Appreciated
`
---
name: Zuvembie -Undead 7
numberencounterd: 1 (1d3)
alignment: CE
size: M
movement: 50
dexterity: 16
armorclass: 4
hitdice: 7
attackrate: 3/1 (claw/claw/bite)
damage: 1d6+1/1d6+1/1d4+1
savingthrow: 13
morale: 12
experience: 1,110
treasure: E
type: Undead 7
source: Hyperborea 3RD - Referee’s Manual
---
```statblock
layout: Hyperborea 3E Monster
image: imagename
monster: name
Number Encountered: numberencountered
Alignment: alignment
Size: size
Movement: movement
Dexterity: dexterity
Armor Class: armorclass
Hit Dice: hitdice
Attack Rate: attackrate
Damage: damage
Saving Throw: savingthrow
Morale: morale
Experience: experience
Treasure: treasure
Type: type
Source: source
Description: description
Special: special
```

# Zuvembie -Undead 7
**Number Encountered** `=this.numberencounterd`
**Alignment:** `=this.alignment`
**Size:** `=this.size`
**Movement:** `=this.movement`
**Dexterity:** `=this.dexterity`
**Armor Class:** `=this.armorclass`
**Hit Dice:** `=this.hitdice`
**Attack Rate:** `=this.attackrate`
**Damage:** `=this.damage`
**Saving Throw:** `=this.savingthrow`
**Morale:** `=this.morale`
**Experience:** `=this.experience`
**Treasure:** `=this.treasure`
**Type:**`=this.type`
**Source:** `=this.source`

## **Description:** 
The zuvembie is the result of a woman imbibing a black brew. The resultant transformation is a free-willed undead creature: a provocatively beautiful yet frightening zombie, immortal as a vampire and nearly as keen. The zuvembie retains no memory of its past, forever changed and inscrutable in disposition. It cannot speak, but it can vocalize weird and haunting melodies. With its song, it can control small animals and oft maintains unlikely combinations of such creatures in its vicinity (typically old towers, abandoned farms, ruins, caves, and the like).
Zuvembies oft walk naked in the moonlight, staring at the stars through cadaverous eyes and pondering inhuman thoughts; too, one will stand for hours or days in the hollow of a tree, in the closet of an old house, or under a stair. When opportunity presents itself, it sings its enchanting song to lure a man to its cold embrace, taking him as a lover and then eating his entrails. Zuvembies are renowned for animating their lifeless lovers shortly after killing them and controlling their corpses like puppets.

### **Specials:** 
* Climb: Climb on 11-in-12 chance of success, at full movement rate.
* Diseased Bite: Bite inflicts disease (reverse of cure disease spell).
* Haunting Song: Can sing a haunting song
that charms any human males within 60 feet who hear it, unless sorcery saves are made; the saving throws are modified by willpower adjustment, if applicable. Charmed individuals stand dumbly and gaze into the eyes of the zuvembie, allowing the beast to exact its grisly will.
* Immunities: Immune to cold, fear, paralysis, and poison. Immune to mundane weapons; harmed only by silver or magical weapons.
* Spells: Can cast the following spells at will (though only one at a time): danse macabre, darkness (through which it can see), witch fire.
* Summon Animals: Once per week, can summon 3d6 small animals. Typically summoned creatures to include bats, gulls, rats, ravens, or squirrels. Each summoning conjures one animal type that remains in the zuvembie’s vicinity for 1d4 weeks, warning and protecting their master.`
`

Here is the .json code for the Custom Hyperborea 3E Monster Layout I created:

`{"blocks":[{"type":"inline","id":"1a0858aafb2b","properties":[],"hasRule":true,"nested":[{"type":"group","id":"cb695b49cb99","properties":["name","size","type","subtype","alignment"],"nested":[{"type":"heading","id":"29c98b8a399b","properties":["Monster"],"conditioned":false,"fallback":"-","hasRule":true},{"type":"subheading","id":"593a79d9cbca","properties":["NumberEncountered","Alignment","Size","Type"],"conditioned":false,"separator":" ","fallback":"-"}],"conditioned":true},{"type":"image","id":"086bea8a594a","properties":["image"],"fallback":"-","conditioned":true}]},{"type":"group","id":"4b99c8f95bda","properties":["ac","hp","speed"],"nested":[{"type":"inline","id":"2ab959da39ea","properties":[],"nested":[{"type":"property","id":"48f968791a29","display":"Movement","properties":["movement"],"conditioned":false,"fallback":"-"},{"type":"property","id":"6b9bdaaae89a","properties":["Dexterity"],"fallback":"-","display":"Dexterity"}]},{"type":"inline","id":"68fb485999a8","properties":[],"nested":[{"type":"property","id":"080898aa1bda","properties":["ArmorClass"],"display":"Armor Class","conditioned":false,"fallback":"-"},{"type":"property","id":"99a85b0929f9","properties":["HitDice"],"fallback":"-","display":"Hit Dice"}]},{"type":"inline","id":"e9594b481a88","properties":[],"nested":[{"type":"property","id":"e8e83b3a5988","properties":["AttackRate"],"fallback":"-","display":"Attack Rate"},{"type":"property","id":"4be82b88eaba","properties":["Damage"],"fallback":"-","display":"Damage"}]},{"type":"inline","id":"3b192b58b81a","properties":[],"nested":[{"type":"property","id":"381bab087b69","properties":["SavingThrow"],"fallback":"-","display":"Saving Throw"},{"type":"property","id":"bb596978a8ea","properties":["Morale"],"fallback":"-","display":"Morale"}]},{"type":"inline","id":"0afb5aab896b","properties":[],"nested":[{"type":"property","id":"cafb1919da88","properties":["Experience"],"fallback":"-","display":"Experience"},{"type":"property","id":"d8a99a38784a","properties":["Treasure"],"fallback":"-","display":"Treasure"}]}],"hasRule":true,"conditioned":true},{"type":"text","id":"a8fb295b88f9","properties":["Description"],"text":null,"fallback":"-","heading":"Description","hasRule":true},{"type":"text","id":"7a7829cb3b18","properties":["Special"],"text":null,"fallback":"-","hasRule":true,"heading":"Special","conditioned":true},{"type":"text","id":"d9c8b90b0afa","properties":["Source"],"text":null,"fallback":"-","heading":"Source","hasRule":true}],"name":"Hyperborea 3E Monster"}
`

**Comment by BobHood** [comment]
Note the output is as follows:

FrontMatter Block
StatBlock
Template (using Frontmatter's data)

Here is an Empty copy of the Handlebar Template (statblock inserted at the same location) pasted in as text rather than code block hoping it shows all the characters.

---
name: {{Monster}}
layout: Hyperborea 3E Monster
image: {[[imagename]]}
numberencounterd: {{NoEncountered}}
alignment: {{Alignment}}
size: {{Size}}
movement: {{Movement}}
dexterity: {{Dexterity}}
armorclass: {{ArmorClass}}
hitdice: {{HitDice}}
attackrate: {{AttackRate}}
damage: {{Damage}}
savingthrow: {{SavingThrow}}
morale: {{Morale}}
experience: {{Experience}}
treasure: {{Treasure}}
type: {{Type}}
source: {{Source}}
description: {{Description}}
special: {{Special}}
statblock: true
---

```statblock
layout: Hyperborea 3E Monster
image: `=this.imagename`
monster: `=this.name`
Number Encountered: `=this.numberencounterd`
Alignment: `=this.alignment`
Size: `=this.size`
Movement: `=this.movement`
Dexterity: `=this.dexterity`
Armor Class: `=this.armorclass`
Hit Dice: `=this.hitdice`
Attack Rate: `=this.attackrate`
Damage: `=this.damage`
Saving Throw: `=this.savingthrow`
Morale: `=this.morale`
Experience: `=this.experience`
Treasure: `=this.treasure`
Type:`=this.type`
Source: `=this.source`
Description: `=this.description`
Special: `this.special`
```

# {{Monster}}
![[`this.imagename`]]
**Number Encountered:** `=this.numberencounterd`
**Alignment:** `=this.alignment`
**Size:** `=this.size`
**Movement:** `=this.movement`
**Dexterity:** `=this.dexterity`
**Armor Class:** `=this.armorclass`
**Hit Dice:** `=this.hitdice`
**Attack Rate:** `=this.attackrate`
**Damage:** `=this.damage`
**Saving Throw:** `=this.savingthrow`
**Morale:** `=this.morale`
**Experience:** `=this.experience`
**Treasure:** `=this.treasure`
**Type:**`=this.type`
**Source:** `=this.source`

## **Description:** 
{{Description}}

### **Specials:** 
{{Special}}

**Comment by sigrunixia** [comment]
Technically, you are not pulling front matter into Statblocks with this setup, but pulling frontmatter into the note with Dataview.

For a true statblock setup, your front matter should be like so:

```yaml
---
layout: Hyperborea 3E Monster
statblock: true
name: Zuvembie -Undead 7
numberencounterd: 1 (1d3)
alignment: CE
size: M
movement: 50
dexterity: 16
armorclass: 4
hitdice: 7
attackrate: "3/1 (claw/claw/bite)"
damage: "1d6+1/1d6+1/1d4+1"
savingthrow: 13
morale: 12
experience: 1110
treasure: E
type: Undead 7
source: "Hyperborea 3RD - Referee’s Manual"
---
``` 

and then in the note, you have the following

````yaml
```statblock
monster: Zuvembie -Undead 7
```
````

We're trending away from front matter statblocks, partly because of changes coming up that will vastly overhaul how metadata is handled. Below is the preferred method going forward, as dataview can access the bestiary so there is no need for front matter statblocks.

#### Frontmatter

```yaml
---
alias:
statblock: inline
tags:
---

```

#### Inside the note...

````yaml
```statblocks
layout: Hyperborea 3E Monster
name: Zuvembie -Undead 7
numberencounterd: 1 (1d3)
alignment: CE
size: M
movement: 50
dexterity: 16
armorclass: 4
hitdice: 7
attackrate: "3/1 (claw/claw/bite)"
damage: "1d6+1/1d6+1/1d4+1"
savingthrow: 13
morale: 12
experience: 1110
treasure: E
type: Undead 7
source: "Hyperborea 3RD - Referee’s Manual"
```
````
